graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/strings/KeyValue.java

```diff
@@ -139,6 +149,20 @@ static CharMatcher withQuoteHandling(CharMatcher charMatcher) {
                     .and(charMatcher);
         }
 
+        /**
+         * An implementation that doesn't split when the given delimiter char matcher appears in double or single quotes
+         * or after the escape char '\'.
+         *
+         * @param charMatcher the char matcher
+         * @return a char matcher that can handle double, single quotes, and escape characters
+         */
+        static CharMatcher withQuoteAndEscapeHandling(CharMatcher charMatcher) {
+            return new DelimiterCharMatcher('"')
+                    .and(new DelimiterCharMatcher('\''))
```

Comment:
Super-nit: These two lines could be extracted to a `BASE_MATCHER` const and also used in the method above since this code just adds-on the not-escaped matcher to the base.

```
private static final CharMatcher BASE_MATCHER = new DelimiterCharMatcher('"').and(new DelimiterCharMatcher('\''));
```

Reply:
Moving that code out to a variable actually breaks functionality in the existing keyValue tests. Not exactly sure how, but leaving as is to avoid that. @danotorrey

Reply:
Thanks for giving it a go!